src/pydicom/pixels/decoders/base.py

```diff
@@ -0,0 +1,1980 @@
+                    "expected number for uncompressed data - check you have "
+                    "set the correct transfer syntax"
+                )
+
+            return
+
+        # Correct for the trailing NULL byte padding for odd length data
+        padded = expected + expected % 2
+        if actual < padded:
+            if actual != expected:
+                raise ValueError(
+                    f"The number of bytes of pixel data is less than expected "
+                    f"({actual} vs {padded} bytes) - the dataset may be "
+                    "corrupted, have an invalid group 0028 element value, or "
+                    "the transfer syntax may be incorrect"
+                )
+        elif actual > padded:
+            if self.photometric_interpretation == PI.YBR_FULL_422:
+                # PS 3.3, Annex C.7.6.3
+                ybr_length = expected // 2 * 3
+                if actual >= ybr_length + ybr_length % 2:
+                    raise ValueError(
+                        "The number of bytes of pixel data is a third larger "
+                        f"than expected ({actual} vs {expected} bytes) which "
+                        "indicates the set photometric interpretation "
+                        "'YBR_FULL_422' is incorrect"
+                    )
+
+            # PS 3.5, Section 8.1.1
+            warn_and_log(
+                f"The pixel data is {actual} bytes long, which indicates it "
+                f"contains {actual - expected} bytes of excess padding to "
+                "be removed"
+            )
+
+    def validate_options(self) -> None:
+        """Validate the supplied options to ensure they meet minimum requirements."""
+        # Minimum required
+        required_keys = [
+            "bits_allocated",
+            "bits_stored",
+            "columns",
+            "number_of_frames",
+            "photometric_interpretation",
+            "pixel_keyword",
+            "rows",
+            "samples_per_pixel",
+        ]
+        missing = [k for k in required_keys if k not in self._opts]
+        if missing:
+            raise AttributeError(f"Missing expected options: {', '.join(missing)}")
+
+        if not 1 <= self.bits_allocated <= 64:
+            raise ValueError("Bits allocated must be in the range (1, 64)")
+
+        if self.bits_allocated != 1 and self.bits_allocated % 8:
+            raise ValueError("Bits allocated must be 1 or a multiple of 8")
+
+        if not 1 <= self.bits_stored <= self.bits_allocated <= 64:
+            raise ValueError(
+                "Bits stored must be in the range (1, 64) and no greater than "
+                "bits allocated"
+            )
+
+        if not 0 < self.columns <= 2**16 - 1:
+            raise ValueError("Columns must be in the range (1, 65535)")
+
+        if self.number_of_frames < 1:
+            raise ValueError("Number of frames must be greater than or equal to 1")
+
```

Comment:
Minor suggestion - in all these sanity checks, it would be useful to the user to show the actual values found.  E.g. "Columns value 66000 is not within required range (1, 65535)."  There is that kind of info in some of the messages, and I know it would be a lot of work to add them all, so perhaps along with any future updates, could maybe could do a few others that are more likely to happen, or most useful to have the number at hand if they did.

Reply:
Done